### PR TITLE
CTRE Simulation -> core

### PR DESCRIPTION
**Pull request by YHerm**

### CODE REVIEW

**Review by bossyshrimp68 — APPROVED** [review]
תפתור את הקונפליקטים ב keyboard

**Review by bossyshrimp68 — DISMISSED** [review]

File: src/main/java/frc/robot/hardware/phoenix6/motor/simulation/TalonFXSimulation.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		configuration.Feedback.FeedbackSensorSource = FeedbackSensorSourceValue.RotorSensor;
+		if (configuration.Feedback.RotorToSensorRatio != 1.0) {
+			configuration.Feedback.SensorToMechanismRatio *= configuration.Feedback.RotorToSensorRatio;
+			configuration.Feedback.RotorToSensorRatio = 1.0;
+		}
+		talonFXWrapper.applyConfiguration(configuration);
+	}
+
+	public void updateMotor() {
+		if (mechanismSimulation == null) {
```
Comment (hillelv) [inline src/main/java/frc/robot/hardware/phoenix6/motor/simulation/TalonFXSimulation.java:35]: if != null do

Comment (drorta) [reply src/main/java/frc/robot/hardware/phoenix6/motor/simulation/TalonFXSimulation.java]: Why?

**Review by hillelv — DISMISSED** [review]

**Review by bossyshrimp68 — DISMISSED** [review]

**Review by drorta — DISMISSED** [review]

File: src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXMotor.java [hunk 2]
```diff
@@ -1,27 +1,62 @@
...
+		SysIdRoutine.Config sysidConfig,
+		MechanismSimulation simulation
+	) {
 		super(logPath);
-		this.motor = motor;
+		this.motor = Robot.ROBOT_TYPE.isSimulation() ? new TalonFXWrapper(deviceID.ID()) : new TalonFXWrapper(deviceID);
+		if (!motor.applyConfiguration(configuration, APPLY_CONFIG_RETRIES).isOK()) {
+			new Alert(Alert.AlertType.ERROR, getLogPath() + "ConfigurationFailed").report();
+		}
+		this.talonFXSimulation = Robot.ROBOT_TYPE.isSimulation() && simulation != null
```
Comment (Sh-Pe) [inline src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXMotor.java:38]: why would you pass a null simulation?

Comment (YHerm) [reply src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXMotor.java:38]: If u don’t want to run it, like there is no reason to run simulation code (that takes more run time) when running on real robot for example. The simulation value is being checked before the update if it’s null so it should not make any problem

**Review by hillelv — APPROVED** [review]

**Review by Sh-Pe — APPROVED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Pull request merged**